contributing_to_docs/doc_guidelines.adoc

```diff
@@ -457,6 +457,72 @@ If it makes more sense in context to refer to the major version of the product i
+
+You can use ifdef and ifeval statements to control the way content displays in different distributions and assemblies.
+
+NOTE: You can nest conditional statements that involve distribution and assembly context, but you must ensure that you close the if statements correctly.
+
+Because we maintain separate branches for each OpenShift Container Platform version, do not use if statements that are based on product version to vary content.
+
+[id="conditionals-for-distributions"]
+=== Conditionals for distributions
+
+Use ifdef and ifndef statements to control content based on distribution, as described in the previous section. For example, the following example renders differently in OKD and OpenShift Container Platform:
+
+----
+\ifdef::openshift-origin[]
+You can link:https://www.keycloak.org/docs/latest/server_admin/index.html#openshift[configure a Keycloak] server as an OpenID
+Connect identity provider for {product-title}.
+\endif::[]
+
+\ifdef::openshift-enterprise[]
+You can
+link:https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/[configure Red Hat Single Sign-On]
+as an OpenID Connect identity provider for {product-title}.
+\endif::[]
+----
+
+In OKD, this section renders as the following text:
+
+> You can link:https://www.keycloak.org/docs/latest/server_admin/index.html#openshift[configure a Keycloak] server as an OpenID
+Connect identity provider for OKD.
+
+In OpenShift Container Platform, this section renders as the following text:
+
+> You can
+link:https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/[configure Red Hat Single Sign-On]
+as an OpenID Connect identity provider for OpenShift Container Platform.
+
+
+[id="conditionals-for-assemblies"]
+=== Conditionals for different assemblies
+
+Use a combination of ifdef and ifeval statements to control content that needs to vary between assemblies. These conditional statements rely on a combination of the the context attribute for each assembly and specific temporary attributes within each module to control content.
+
+The following sample shows a simple example. In the assembly that contains the `context` attribute `updating-restricted-network-cluster`, an extra paragraph is displayed.
+
+----
+\ifeval::["{context}" == "updating-restricted-network-cluster"]
+:restricted:
+\endif::[]
+
+...
+
+\ifdef::restricted[]
+If you are upgrading a cluster in a restricted network, install the `oc` version that you plan to upgrade to.
+\endif::restricted[]
+
+...
+
+\ifeval::["{context}" == "updating-restricted-network-cluster"]
+:!restricted:
+\endif::[]
+----
+
+Note that you must set and unset each temporary attribute that you introduce to an assembly. Use the temporary attributes in the applicable ifdef and ifndef statements to vary text between the assemblies. The preceeding example uses `restricted` as the temporary attribute to display an additional paragraph for the assembly with the `updating-restricted-network-cluster` context attribute.
```

Comment:
@bergerhoffer, is this clearer?

Reply:
@kalexand-rh sorry for the delay. Yes this is much clearer thanks!

Reply:
@bergerhoffer, I lost track of it for longer. :) Thank you!